Utils/ObjectPool.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DUCK.Pooling
+{
+	public abstract class PoolableComponent : MonoBehaviour
+	{
+		protected internal virtual void OnAddedToPool()
```

Comment:
I'm not sure we need internal here, duck will currently always exist in the project's `AssemblyCSharp` assembly

Reply:
Done